speed-dreams/source-2.2.3/src/modules/graphic/ssggraph/grboard.cpp

```diff
@@ -1550,6 +1555,43 @@ void cGrBoard::refreshBoard(tSituation *s, const cGrFrameInfo* frameInfo,
   if (trackMap) {
       trackMap->display(currCar, s, 0, 0, rightAnchor, TOP_ANCHOR);
   }
+
+}
+
+/// @brief Displays the texture stored in interventionTexture (which is loaded in grInitBoardCar, move this later)
+void cGrBoard::grDispIntervention() 
+{
+    // Dimensions of the icon on the screen (will be put in XML settings file later)
+    float iconWidth = 100;
+    float iconHeight = 100;
+
+    // Duplicate the current matrix and enable opengl settings.
+    glPushMatrix();
+    glEnable(GL_BLEND);
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    glEnable(GL_TEXTURE_2D);
+    glTexEnvf(GL_TEXTURE_ENV, GL_TEXTURE_ENV_MODE, GL_REPLACE);
+
+    // Translate the opengl matrix to the position on the screen where we want to display the texture, and load the texture.
+    glTranslatef(1.5 * centerAnchor - 0.5 * iconWidth, BOTTOM_ANCHOR, 0);
+    if (interventionTexture) {
+        glBindTexture(GL_TEXTURE_2D, interventionTexture->getTextureHandle());
+    }
+
+    // Draw the texture as a Triangle Strip. 
+    // glTexCoord2f defines point of the texture that you take (0-1).
+    // glVertex2f then defines where to place this on the screen (relative to the current matrix)
+    glBegin(GL_TRIANGLE_STRIP);
+    glColor4f(1.0, 1.0, 1.0, 0.0);
+    glTexCoord2f(0.0, 0.0); glVertex2f(0, 0);
+    glTexCoord2f(0.0, 1.0); glVertex2f(0, iconHeight);
+    glTexCoord2f(1.0, 0.0); glVertex2f(iconWidth, 0);
+    glTexCoord2f(1.0, 1.0); glVertex2f(iconWidth, iconHeight);
+    glEnd();
```

Comment:
If the texture is not loaded correctly it will not bind a texture but still draw, should the if be a guard for the entire function?

Reply:
fixed by adding a guarding if statement